spec/tic_tac_toe/game_spec.clj

```diff
@@ -2,14 +2,27 @@
   (:require [speclj.core :refer :all]
             [tic-tac-toe.board-spec :refer [empty-board]]
             [tic-tac-toe.game :refer :all]
-            [tic-tac-toe.marks :refer :all]))
+            [tic-tac-toe.marks :refer :all]
+            [tic-tac-toe.messages :refer [draw-message
+                                          winner-message]]))
 
-(describe "switch-player"
-          (let [player-one {:type :human :mark "X"} player-two {:type :human :mark "O"}]
-            (it "returns player one mark if currently player two"
-                (should= player-one
-                         (switch-player player-two player-one player-two)))
+(def players [{:type :human :mark "X"} {:type :human :mark "O"}])
 
-            (it "returns player two mark if currently player one"
-                (should= player-two
-                         (switch-player player-one player-one player-two)))))
+(describe "run-game"
+          (it "does not throw error when game won"
+              (let [won-board [player-one-mark player-one-mark "3"
+                               player-two-mark player-two-mark "6"
+                               "7" "8" "9"]
+                    winning-choice "3"
+                    delay-time 0]
+                (should= nil
+                         (with-in-str winning-choice (run-game players won-board delay-time)))))
```

Comment:
Interesting test - you're checking that when a valid action is taken it doesn't throw an error. I would approach testing the valid use cases against the expected outcome. I think this highlights that run-game is a hard function to write a meaningful test for.

Reply:
yes agreed - I initially tried to test the print output - but the problem was isolating the final winning message as there are various other messages printed before the winning message, which I wished to ignore... so this was the only one I could come up with - in java the board is an object that you can potentially pull up to look at - not so in clojure, of course